### ISSUE TITLE
When the server is unavailable, provide a way to handle promise rejection

### ISSUE THREAD

**Issue by fabianTMC** [atroo/node-weedfs#5]
When I try to store a file when the server is offline, it logs the following error:

```
(node:16587) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: connect ECONNREFUSED 127
.0.0.1:9333
```

This is my code:
```
weedfs.write(file.stream)
.catch(function(err) {
    cb(err, null);
})
.then((fileInfo) => {
    cb(null, fileInfo.fid);
})
```

I've gone through the documentation but there is nothing mentioned about this. Any help would be appreciated.

**Comment by phal0r** [comment]
Hi @fabianTMC ,

thanks for reporting this. You are right, the Promise chain was broken during assignment requests and errors could not be catched. I refactored this and added a test for failing connections. Please have a look at v1.3.0 if it fits your use case.

**Comment by atroo** [comment]
Fixed with 1.3.0
